source/jquery.flot.js

```diff
@@ -2453,14 +2453,28 @@ Licensed under the MIT license.
         };
 
         function computeBarWidth(series) {
+            var xValues = [];
             var pointsize = series.datapoints.pointsize, minDistance = Number.MAX_VALUE,
                 distance = series.datapoints.points[pointsize] - series.datapoints.points[0] || 1;
 
             if (isFinite(distance)) {
                 minDistance = distance;
             }
-            for (var j = pointsize; j < series.datapoints.points.length - pointsize; j += pointsize) {
-                distance = Math.abs(series.datapoints.points[pointsize + j] - series.datapoints.points[j]);
+
+            for (var j = 0; j < series.datapoints.points.length; j += pointsize) {
+                if (isFinite(series.datapoints.points[j]) && series.datapoints.points[j] !== null) {
+                    xValues.push(series.datapoints.points[j]);
+                }
+            }
+
+            function onlyUnique(value, index, self) { 
+                return self.indexOf(value) === index;
+            }
+
+            xValues = xValues.filter( onlyUnique );
+
+            for (var j = 1; j < xValues.length; j++) {
+                distance = Math.abs(xValues[j] - xValues[j - 1]);
```

Comment:
So, in our discussion I was under the impression that xValues was sorted AND filtered for unique values, but this only seems to be filtered for unique values. Sorting still seems important for the following scenario:

data input:
{ {3, 5, 0}, {8, 2, 0}, {1, 2, 0} } (So, x-values of 3, 8, 1).

The algorithm as written will come up with a distance of 5. Whereas with sorting it will come up with a distance of 2.

Reply:
In the end I decided that eliminating duplicates was the way to go. That is what the category axis itself does. Sorting yielded incorrect results.

Reply:
After reviewing the behavior described above, Andrew and I are in agreement that the preferred behavior requires the x-values to be sorted. Had to update a test as a result of that.